File: android/app/src/main/java/com/by/android/data/model/Job.kt
package com.by.android.data.model

import androidx.compose.ui.graphics.Color

data class Job(
    val name: String,
    val url: String,
    val color: String? = null,
    val lastBuild: BuildReference? = null,
    val lastSuccessfulBuild: BuildReference? = null,
    val lastFailedBuild: BuildReference? = null,
    val lastCompletedBuild: BuildReference? = null,
    val buildable: Boolean? = null,
    val healthReport: List<HealthReport>? = null
) {
    val status: JobStatus
        get() = when (color) {
            "blue", "blue_anime" -> JobStatus.SUCCESS
            "red", "red_anime" -> JobStatus.FAILURE
            "yellow", "yellow_anime" -> JobStatus.UNSTABLE
            "grey", "grey_anime", "disabled", "disabled_anime" -> JobStatus.DISABLED
            "aborted", "aborted_anime" -> JobStatus.ABORTED
            "notbuilt", "notbuilt_anime" -> JobStatus.NOT_BUILT
            else -> JobStatus.UNKNOWN
        }
    
    val isBuilding: Boolean
        get() = color?.contains("_anime") == true
    
    val healthScore: Int
        get() = healthReport?.firstOrNull()?.score ?: 0
}

data class BuildReference(
    val number: Int,
    val url: String
)

data class HealthReport(
    val description: String? = null,
    val score: Int = 0
)

enum class JobStatus {
    SUCCESS,
    FAILURE,
    UNSTABLE,
    DISABLED,
    ABORTED,
    NOT_BUILT,
    UNKNOWN;
    
    val statusColor: Color
        get() = when (this) {
            SUCCESS -> Color(0xFF4CAF50)
            FAILURE -> Color(0xFFF44336)
            UNSTABLE -> Color(0xFFFF9800)
            DISABLED -> Color(0xFF9E9E9E)
            ABORTED -> Color(0xFF9E9E9E)
            NOT_BUILT -> Color(0xFF9E9E9E)
            UNKNOWN -> Color(0xFF9E9E9E)
        }
}

// API Response wrapper
data class JobsResponse(
    val jobs: List<Job>
)

File: ios/ios/Models/Job.swift
//
//  Job.swift
//  ios
//
//  Jenkins job model
//

import Foundation
import SwiftUI

struct Job: Codable, Identifiable, Equatable, Hashable {
    let name: String
    let url: String
    let color: String?
    let lastBuild: BuildReference?
    let lastSuccessfulBuild: BuildReference?
    let lastFailedBuild: BuildReference?
    let lastCompletedBuild: BuildReference?
    let buildable: Bool?
    let healthReport: [HealthReport]?
    
    var id: String { name }
    
    var status: JobStatus {
        guard let color = color else { return .unknown }
        switch color {
        case "blue", "blue_anime":
            return .success
        case "red", "red_anime":
            return .failure
        case "yellow", "yellow_anime":
            return .unstable
        case "grey", "grey_anime", "disabled", "disabled_anime":
            return .disabled
        case "aborted", "aborted_anime":
            return .aborted
        case "notbuilt", "notbuilt_anime":
            return .notBuilt
        default:
            return .unknown
        }
    }
    
    var isBuilding: Bool {
        color?.contains("_anime") ?? false
    }
    
    var healthScore: Int {
        healthReport?.first?.score ?? 0
    }
}

struct BuildReference: Codable, Equatable, Hashable {
    let number: Int
    let url: String
}

struct HealthReport: Codable, Equatable, Hashable {
    let description: String?
    let score: Int
}

enum JobStatus: String, CaseIterable {
    case success
    case failure
    case unstable
    case disabled
    case aborted
    case notBuilt
    case unknown
    
    var color: Color {
        switch self {
        case .success: return .green
        case .failure: return .red
        case .unstable: return .yellow
        case .disabled: return .gray
        case .aborted: return .gray
        case .notBuilt: return .gray
        case .unknown: return .gray
        }
    }
    
    var iconName: String {
        switch self {
        case .success: return "checkmark.circle.fill"
        case .failure: return "xmark.circle.fill"
        case .unstable: return "exclamationmark.triangle.fill"
        case .disabled: return "pause.circle.fill"
        case .aborted: return "stop.circle.fill"
        case .notBuilt: return "circle.dashed"
        case .unknown: return "questionmark.circle.fill"
        }
    }
}

// MARK: - API Response Wrappers

struct JobsResponse: Codable {
    let jobs: [Job]
}

